dojo/tools/twistlock/parser.py

```diff
@@ -1,16 +1,118 @@
+import io
+import csv
 import json
+import hashlib
+
 from dojo.models import Finding
 
 
-class TwistlockParser(object):
-    def __init__(self, json_output, test):
 
-        tree = self.parse_json(json_output)
 
+class TwistlockCSVParser(object):
+
+    @staticmethod
+    def get_field_from_row(row, column):
+        if row[column] is not None:
+            return row[column]
+        else:
+            return None
+
+    def parse_issue(self, row, test):
+
+        if not row:
+            return None
+
+        id_column = 4
+        line_column = 5
+        code_line_column = 6
+        vulnerability_id_column = 10
+        severity_column = 13
+        package_name_column = 14
+        package_version_column = 16
+        cvss_column = 18
+        fix_status_column = 19
+        description_column = 20
+
+        if self.get_field_from_row(row, vulnerability_id_column) is None:
+            data_vulnerability_id = ''
+        else:
+            data_vulnerability_id = self.get_field_from_row(row, vulnerability_id_column)
+
+        if self.get_field_from_row(row, package_version_column) is None:
+            data_package_version = ''
+        else:
+            data_package_version = self.get_field_from_row(row, package_version_column)
+
+        if self.get_field_from_row(row, fix_status_column) is None:
+            data_fix_status = ''
+        else:
+            data_fix_status = self.get_field_from_row(row, fix_status_column)
+
+        data_package_name = self.get_field_from_row(row, package_name_column)
+        data_id = self.get_field_from_row(row, id_column)
+        if self.get_field_from_row(row, severity_column) == '':
+            data_severity = 'Info'
+        else:
+            data_severity = self.get_field_from_row(row, severity_column).capitalize()
+
+        data_cvss = self.get_field_from_row(row, cvss_column)
+        data_description = self.get_field_from_row(row, description_column)
+        finding = Finding(
+            title=data_vulnerability_id + ": " + data_package_name + " - " + data_package_version,
+            cve=data_vulnerability_id,
+            test=test,
+            severity=data_severity,
+            description=data_description + "<p> Vulnerable Package: " +
```

Comment:
Does this snippet of HTML can lead to a XSS when it gets injected into the view?

Reply:
I think all the fields which are read form the CSV file and stored directly into the fields of the Finding are prone to XSS. They should be sanitized.

Reply:
Used https://docs.djangoproject.com/en/3.0/ref/utils/#django.utils.html.escape to sanitize the input data